runtime/src/feature.rs

```diff
@@ -0,0 +1,42 @@
+use solana_sdk::{account::Account, clock::Slot};
+
+solana_sdk::declare_id!("Feature111111111111111111111111111111111111");
+
+/// The `Feature` struct is the on-chain representation of a runtime feature.
+///
+/// Feature activation is accomplished by:
+/// 1. Activation is requested by the feature authority, who issues a transaction to create the
+///    feature account.  The newly created feature account will have the value of
+///    `Feature::default()`
+/// 2. When the next epoch is entered the runtime will check for new activation requests and
+///    active them.  When this occurs, the activation slot is recorded in the feature account
+///
+#[derive(Default, Debug, Serialize, Deserialize)]
+pub struct Feature {
+    pub activated_at: Option<Slot>,
+}
+
+impl Feature {
+    pub fn size_of() -> usize {
+        bincode::serialized_size(&Self {
+            activated_at: Some(Slot::MAX),
+        })
+        .unwrap() as usize
+    }
+    pub fn from_account(account: &Account) -> Option<Self> {
+        if account.owner != id() {
+            None
+        } else {
+            bincode::deserialize(&account.data).ok()
+        }
+    }
+    pub fn to_account(&self, account: &mut Account) -> Option<()> {
+        bincode::serialize_into(&mut account.data[..], self).ok()
+    }
+    pub fn create_account(&self, lamports: u64) -> Account {
+        let data_len = Self::size_of().max(bincode::serialized_size(self).unwrap() as usize);
+        let mut account = Account::new(lamports, data_len, &id());
+        self.to_account(&mut account).unwrap();
+        account
+    }
```

Comment:
(Sorry for further post-merge review comments; I'm finishing this off)

nits: According to this way of [creating new `Feature` accounts](https://github.com/solana-labs/solana/pull/12376/files#diff-2b291ef836c6fc9eeaafd31e6a38d639R256) (skipping special system instruction for `Feature`), this is assuming `Feature::size_of()`-sized zero byte arrays should be always to be deserialized as `Feature { activated_at: None }` successfully? This is sensible bet. However, this could be change `bincode` version up, or switch to completely new other deserialization implementation (protobuf, borsh).

How about writing a tiny unit-test to assume that? Or am I worrying too much?

```
  #[test]
  fn test_deserialize_blank_new_account() {
      let just_initialized = Account::new(42, Feature::size_of(), &id());
      assert_eq!(Feature::from_account(&just_initialized), Some(Feature{activated_at: None}));
  }
```

Reply:
Done in https://github.com/solana-labs/solana/pull/12524